.github/workflows/quality-check.yaml

```diff
@@ -20,34 +20,56 @@ on:
     types:
       - created
 
-
 jobs:
   check-and-test:
     runs-on: ubuntu-latest
     strategy:
       fail-fast: false
       matrix:
-        python-version: [3.7, 3.8]
-        poetry-version: [1.1.6]
+        python-version: [3.10.12]
+        poetry-version: [1.1.12]
     name: Python-${{ matrix.python-version }}
     steps:
       - uses: actions/checkout@v2
       - uses: actions/setup-python@v2
         with:
           python-version: ${{ matrix.python-version }}
-      # ------------ Install poetry
+
       - name: Setup pip/poetry
         run: |
           pip install -U pip poetry twine
           poetry config virtualenvs.create false
-      # ------------ install tools
+
+      - name: Install Python dependencies
+        run: poetry install
+
+      - name: Configure environment variables for CMake
+        run: |
+          echo "PYTHON_BIN=$(which python)" >> $GITHUB_ENV
+
+      - name: Verify TensorFlow installation
+        run: |
+          python -c "import tensorflow as tf; print(tf.__version__)"
+
       - name: Install building tools
         run: |
-          sudo apt-get install build-essential
-          sudo apt-get install cmake g++-11
-      # ------------ build and install package
+          sudo apt-get update
+          sudo apt-get install -y build-essential cmake g++-14
+
+      - name: Configure and Build C++ Library
+        run: |
+          mkdir -p banded_matrices/build
+          cd banded_matrices/build
+          cmake .. -DCMAKE_BUILD_TYPE=Release
```

Comment:
I would be tempted to use the `-B` option of `cmake`: https://cmake.org/cmake/help/latest/manual/cmake.1.html#generate-a-project-buildsystem